File: CompositeVisitor/CompositeVisitor_java/CompositeShape.java
public class CompositeShape implements IShape {
  java.util.LinkedList<IShape> m_shapes;
  
  public CompositeShape() {
    m_shapes = new java.util.LinkedList<IShape>();
  }
  
  public void add(IShape a_shape) {
    m_shapes.add(a_shape);
  }
  
  public float getArea() {
    float total = 0;
    
    for(IShape s : m_shapes) {
      total += s.getArea();
    }
    
    return total;
  }
  
  public void accept(IShapeVisitor a_visitor) {
    a_visitor.visitCompositePre(this);
    for(IShape s : m_shapes) {
      s.accept(a_visitor);
    }
    a_visitor.visitCompositePost(this);
  }
}

File: CompositeVisitor/CompositeVisitor_Swift_3.0.1/CompositeVisitor/model/CompositeShape.swift
//
//  CompositeShape.swift
//  CompositeVisitor
//
//  Created by Sunny Buttar on 2016-11-13.
//  Copyright © 2016 Sarpreet Singh. All rights reserved.
//

import Foundation

class CompositeShape: IShape {
    
    private var m_shapes : Array<IShape>;
    
    init() {
        m_shapes = Array<IShape>();
    }
    
    func add(a_shape : IShape) {
        m_shapes.append(a_shape);
    }
    
    func getArea() -> Float {
        var total : Float = 0;
        
        for item in m_shapes {
            total += item.getArea();
        }
        return total;
    }
    
    func accept(a_visitor: IShapeVisitor) {
        a_visitor.visitCompositePre(a_group: self);
        
        for item in m_shapes {
            item.accept(a_visitor: a_visitor);
        }
        a_visitor.visitCompositePost(a_group: self);
    }
}
